lib/rucio/common/stomp_utils.py

```diff
@@ -15,145 +15,415 @@
+
+        try:
+            use_ssl = config_get_bool(config_section, 'use_ssl')
+        except Exception as exc:
+            self._logger(logging.ERROR, "could not find use_ssl in configuration -- please update your rucio.cfg")
+            raise RuntimeError('could not find use_ssl in configuration -- please update your rucio.cfg') from exc
+
+        port = config_get_int(config_section, 'port')
+        vhost = config_get(config_section, 'broker_virtual_host', raise_exception=False)
+        destination = config_get(config_section, "destination")
+        key_file = config_get(config_section, 'ssl_key_file', default=None, raise_exception=False)
+        cert_file = config_get(config_section, 'ssl_cert_file', default=None, raise_exception=False)
+        username = config_get(config_section, 'username', default=None, raise_exception=False)
+        password = config_get(config_section, 'password', default=None, raise_exception=False)
+        nonssl_port = config_get_int(config_section, 'nonssl_port', default=0, raise_exception=False)
+        timeout = config_get_float(config_section, 'timeout', default=None, raise_exception=False)
+        heartbeats = tuple(config_get_list(config_section, 'heartbeats', default=[0., 0.], raise_exception=False))
+        reconnect_attempts = config_get_int(config_section, 'reconnect_attempts', default=100)
+        if use_ssl and (key_file is None or cert_file is None):
+            self._logger(logging.ERROR, "If use_ssl is True in config you must provide both 'ssl_cert_file' "
+                                        "and 'ssl_key_file'")
+            raise RuntimeError("If use_ssl is True in config you must provide both 'ssl_cert_file' and 'ssl_key_file'")
+        if not use_ssl and (username is None or password is None or nonssl_port == 0):
+            self._logger(logging.ERROR, "If use_ssl is False in config you must provide "
+                                        "'username', 'password' and 'nonssl_port'")
+            raise RuntimeError("If use_ssl is False in config you must provide "
+                               "'username', 'password' and 'nonssl_port'")
+        return StompConfig(brokers=self._resolve_host_and_port(brokers, port if use_ssl else nonssl_port),
+                           use_ssl=use_ssl,
+                           port=port, vhost=vhost,
+                           destination=destination, key_file=key_file, cert_file=cert_file,
+                           username=username, password=password, nonssl_port=nonssl_port,
+                           reconnect_attempts_max=reconnect_attempts, timeout=timeout, heartbeats=heartbeats)
+
+    def _resolve_host_and_port(self, fqdns: str | list[str], port: int) -> list[tuple[str, int]]:
+        """
+        Resolve host and port.
+
+        Args:
+            fqdns (str | list[str]): fully qualified domain name(s)
+            port (int): port
+
+        Returns:
+            list[tuple[str, int]]: list of (host, port) tuples.
```

Comment:
Please avoid specifying the argument types in the docstring if they are already present as type annotations; there's no need to duplicate this information and it can cause mismatches between the type annotation and the type specified in the docstring if changes are made down the line